assets/admin/js/globalpayments-admin.js

```diff
@@ -81,6 +83,42 @@
             return $( this.getLiveModeSelector() ).is( ':checked' );
         },
 
+        /**
+         * Checks if gateway setting is enabled
+         *
+         * @returns {*|jQuery}
+         */
+        isEnabled: function() {
+            return $( this.getEnabledGatewaySelector() ).is( ':checked' );
+        },
+
+        /**
+         * Toggle validations when enabled gateway settings
+         */
+        toggleValidations: function () {
+            var button = $('.woocommerce-save-button');
+            if ( this.isEnabled() ) {
+                button.removeAttr( "formnovalidate" );
+            } else {
+                button.attr( "formnovalidate","");
+            }
+        },
+
+        /**
+         * Toggle required settings
+         */
+        toggleRequiredSettings: function () {
+            var enabled = this.isEnabled();
+            var list =  $('.required');
+            list.each(function(){
+                if ($(this).is(':visible')) {
+                    $(this).prop('required', enabled);
```

Comment:
@apetrovici  if we add true instead of using enabled var , we are validating the field when gateway is disabled, we don't want that, so i left this way.